InfluxDbTcpSender + InfluxDbUdpListener

shanielh opened the pull request:
Related #38 

Please review my code and tell me if there's something you want to change.

ngong commented on metrics-influxdb/src/main/java/com/izettle/metrics/influxdb/InfluxDbTcpSender.java:
```diff
@@ -0,0 +1,99 @@
+
+    private void retryConnect() throws IOException {
+        if (tcpSocket == null) {
+            this.tcpSocket = new Socket(hostname, port);
```
There is a findbugs error when using an insecure TCP port
```
[INFO] Unencrypted socket to com.izettle.metrics.influxdb.InfluxDbTcpSender (instead of SSLSocket) [com.izettle.metrics.influxdb.InfluxDbTcpSender] At InfluxDbTcpSender.java:[line 93] UNENCRYPTED_SOCKET
```
But I think this can be added into exclude file. And users have to aware that they are on insecure connections.

shanielh replied:
This should be fine because mostly users should use this to connect to localhost, Also, Currently the telegraf tcp listener does not support ssl (at least not in the [code](https://github.com/influxdata/telegraf/tree/master/plugins/inputs/tcp_listener))

ngong commented on metrics-influxdb/src/main/java/com/izettle/metrics/influxdb/InfluxDbTcpSender.java:
```diff
@@ -0,0 +1,99 @@
+        return influxDbWriteObject.getTags();
+    }
+
+    private void retryConnect() throws IOException {
```
This method will only take effect in the beginning of sending first batch of data points I think. If the socket is disconnected, this method will not throw any exception. But these will throw IOExceptions
```
        OutputStream outputStream = tcpSocket.getOutputStream(); 
        outputStream.write(line.getBytes(UTF_8));
        outputStream.flush();
```
So if you are expecting a auto-reconnect feature. The best way I think should be catching all `IOException` and `SocketTimeoutException` in `writeData`, and then close and connect to the socket again. After given retry times we can consider this is failed and throw another Exception

shanielh commented:
Thanks for the comments, I'll fix the pull request until Saturday.

shanielh commented:
Hi @ngong , I've updated my code and tried it on my machine - It worked even when I stopped telegraf and started it after a while. So it's seems that reconnect is working. Also now the build passes and I've added a comment in the `InfluxDbTcpSender` that says that it uses unsecured connections.

Coming from C#, It's nice that `SocketException` in Java inherits from `IOException` 😄

z0mb1ek commented:
wait for this pr

ngong commented:
@shanielh Brilliant work! I will merge this one later. And refactor the builder for using the new senders

shanielh commented:
@ngong Thanks.
